src/main/java/roomescape/controller/ReservationController.java

```diff
@@ -0,0 +1,49 @@
+package roomescape.controller;
+
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.*;
+import roomescape.domain.Reservation;
+import roomescape.dto.ReservationDto;
+import roomescape.exception.NotFoundReservationException;
+import roomescape.service.ReservationService;
+
+import java.net.URI;
+import java.util.List;
+import java.util.Optional;
+
+@Controller
+@RequiredArgsConstructor
+public class ReservationController {
+
+    private final ReservationService reservationService;
+
+    @GetMapping("/reservation")
+    public String showReservationPage(){
+        return "reservation";
+    }
+
+    @GetMapping("/reservations")
+    public ResponseEntity<List<Reservation>> readReservation(){
+        return ResponseEntity.ok(reservationService.checkReservations());
+    }
+
+    @PostMapping("/reservations")
+    public ResponseEntity<Reservation> createReservation(@Valid @RequestBody ReservationDto reservationDto){
+        Reservation createdReservation = reservationService.addReservation(reservationDto);
+
+        return ResponseEntity.created(URI.create("/reservations/"+createdReservation.getId())).body(createdReservation);
+    }
+
+    @DeleteMapping("/reservations/{reservationId}")
+    public ResponseEntity<Void> deleteReservation(@PathVariable Long reservationId){
+        Optional<Reservation> deleteReservation = reservationService.deleteReservation(reservationId);
+        if (deleteReservation.isEmpty()){
+            throw new NotFoundReservationException();
+        }
```

Comment:
```suggestion
Reservation deletedReservation = reservationService.deleteReservation(reservationId)
                    .orElseThrow(NotFoundReservationException::new);
```

Optional은 내부적으로 다양한 메서드 체이닝 기능을 제공해요. 이를 잘 활용하면, 크게 두 가지 장점이 따라오는 것 같아요.
1. 가독성이 좋아져요. 이는 메서드 체이닝의 장점인 것 같습니다.
2. 확정된 값을 받고, 이후 로직을 안정성있게 전개할 수 있어요.

그런 이유에서, Optional 객체를 직접 받아서 이를 조작하기보다, 메서드 체이닝을 통해 확정된(최소한 NULL 값은 아닌) Reservation 객체 값을 얻어내는 것이 좋을 것 같습니다.

Reply:
이런 방법이!!! 조언 감사합니다 :)